app/src/main/java/com/example/triptracker/view/Navigation.kt

```diff
@@ -103,28 +107,50 @@ class Navigation(val navController: NavHostController) {
   }
 
   // Only use this when navigating to the maps screen
-  fun navigateTo(route: String, id: String) {
+  fun navigateTo(route: String, id: String, isRetry: Boolean = false) {
+    val destination = getTopLevelDestinations().find { it.route == "maps" }!!
     Log.d("Navigation", route + id)
-    navController.navigate("MAPS?id=$id") {
-      currentDestination = getTopLevelDestinations().find { it.route == "maps" }!!
-      // set the id when navigating with the map
-      navController.currentBackStackEntry?.arguments?.putString("id", id)
-      // Pop up to the start destination of the graph to
-      // avoid building up a large stack of destinations
-      // on the back stack as users select items
-      popUpTo(navController.graph.findStartDestination().id) { saveState = false }
-      // Avoid multiple copies of the same destination when
-      // reselecting the same item
-      //      launchSingleTop = true
-      // Restore state when reselecting a previously selected item
-      //      restoreState = true
+    if (connection.isDeviceConnectedToInternet()) {
+      if (isRetry) {
+        Log.d("Navigation", "Successfully reconnected to the internet")
+        // Reset next destination
+        nextDestination = null
+        nextId = null
+        goBack()
+      }
+      navController.navigate("MAPS?id=$id") {
+        currentDestination = destination
+        // set the id when navigating with the map
+        navController.currentBackStackEntry?.arguments?.putString("id", id)
+        // Pop up to the start destination of the graph to
+        // avoid building up a large stack of destinations
+
+        // on the back stack as users select items
+        popUpTo(navController.graph.findStartDestination().id) { saveState = false }
+        // Avoid multiple copies of the same destination when
+        // reselecting the same item
+        //      launchSingleTop = true
+        // Restore state when reselecting a previously selected item
+        //      restoreState = true
```

Comment:
Should those lines be un-commented ? Or maybe later ?

Reply:
I am not the one who commented them, so I left them as-is, but I believe we can remove them